### PR TITLE
HM3D

### PR DESCRIPTION
**Pull request by erikwijmans**
## Motivation and Context

Add support for HM3D to the download script.  This behaves a little differently than other things as 1) the download links require authorization, an API token ID and token key (passed in as a user name/password) and HM3D is quite large so there are many slices of it to allow users to only download what they need.

## How Has This Been Tested

By downloading stuff!  Reviewers, if you want to try this but your account hasn't been enabled send me a ping.

### CODE REVIEW

File: habitat_sim/utils/datasets_download.py [hunk 1]
```diff
@@ -101,6 +120,59 @@ def initialize_test_data_sources(data_path):
         },
     }
 
+    data_sources.update(
+        {
+            f"hm3d_{split}_{data_format}": {
+                "source": "https://api.matterport.com/resources/habitat/hm3d-{split}-{data_format}.tar{ext}".format(
```
Comment (Skylion007) [inline habitat_sim/utils/datasets_download.py:126]: Wouldn't this jsut be cleaner with fstrrings.

**Review by Skylion007 — COMMENTED** [review]
Nit

File: habitat_sim/utils/datasets_download.py (hunk 1 above)
Comment (erikwijmans) [reply habitat_sim/utils/datasets_download.py:126]: IMO no because of the need to add .gz to the obj+mtl tarballs

Comment (Skylion007) [reply habitat_sim/utils/datasets_download.py:126]: f strings can have nested ifs. *shrugs*

**Review by Skylion007 — APPROVED** [review]

**Comment by erikwijmans** [comment]
Landing this so it can go out with conda tonight

**Pull request merged**